File: src/main/java/io/github/kydzombie/link/gui/screen/slot/LinkCardSlot.java
package io.github.kydzombie.link.gui.screen.slot;

import io.github.kydzombie.link.item.LinkCardItem;
import net.minecraft.inventory.Inventory;
import net.minecraft.item.ItemStack;
import net.minecraft.screen.slot.Slot;

public class LinkCardSlot extends Slot {
    public LinkCardSlot(Inventory inventory, int index, int x, int y) {
        super(inventory, index, x, y);
    }

    @Override
    public boolean canInsert(ItemStack stack) {
        return stack != null && stack.getItem() instanceof LinkCardItem;
    }
}

File: src/main/java/com/github/kydzombie/link/slot/LinkCardSlot.kt
package com.github.kydzombie.link.slot;

import com.github.kydzombie.link.Link;
import net.minecraft.container.slot.Slot;
import net.minecraft.inventory.InventoryBase;
import net.minecraft.item.ItemInstance;

public class LinkCardSlot extends Slot {
    public LinkCardSlot(InventoryBase inventory, int invSlot, int x, int y) {
        super(inventory, invSlot, x, y);
    }

    @Override
    public boolean canInsert(ItemInstance itemInstance) {
        return itemInstance != null && itemInstance.getType() == Link.LINK_CARD && itemInstance.getStationNBT().getBoolean("linked");
    }
}
